src/dbus/rauc.rs

```diff
@@ -201,11 +205,26 @@ async fn channel_polling_task(
 
         if let Err(e) = channel.poll(&proxy, slot_status.as_deref()).await {
             warn!(
-                "Failed to fetch update for update channel \"{}\": {}",
-                channel.name, e
+                "Failed to fetch update for update channel \"{}\": {}. Retrying in {}s.",
+                channel.name,
+                e,
+                retry_interval.as_secs()
             );
+
+            if retry_interval < RETRY_INTERVAL_MAX {
+                sleep(retry_interval).await;
```

Comment:
This needs to move outside the if, otherwise we only sleep as long as the interval is smaller than `RETRY_INTERVAL_MAX`.

Reply:
That's actually intentional. The control flow is a bit convoluted here because there are some hairy details that need handling (like stopping the retries if the update channel was disabled in the meantime or polling for updates was disabled alltogether. Or actually updating the internal representation of the update channel to show that no update bundle is available for it after we had some retries).

The idea is that the `continue` in line 222 is not reached if `retry_interval >= RETRY_INTERVAL_MAX`, resulting in the rest of the `while`-loop body being executed. Which will mark the update channel as not having a bundle available and will wait for a normal update polling interval (which can be in the order of days) before making the next attempt.